### ISSUE TITLE
CREATE_DIRS_FAILED on Android

### ISSUE THREAD

**Issue by fgenoese** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#132]
This code on Android:
```
window.plugins.socialsharing.shareViaFacebook(null, 
                                                   image, 
                                                   null,
                                                   function() { 
                                                        console.log("yay!");
                                                    },
                                                    function(err){
                                                        //error callback
                                                        console.log('error in share: '+ err);
                                                    }
                                                  );
```

gives me:
"error in share: CREATE_DIRS_FAILED"

image is an image created from a canvas using toDataURL("image/jpeg");

The same code works on iOS - known limitation?

**Comment by matuskopo** [comment]
Hi,

I'm using this code:
```javascript
window.plugins.socialsharing.share('The best game in the world', 'My Test App', 'https://www.google.sk/images/srpr/logo11w.png', 'http://www.google.sk', function(){alert("ok")}, function(e){alert("error: " + e)});
```

and when I run it on HTC Desire X (Android 4.1.1), errorCallback reports CREATE_DIRS_FAILED error.
On other devices it works fine (Nexus tablet, Samsung Galaxy).

Any sugestions?
Thank you

**Comment by EddyVerbruggen** [comment]
Hmm, does that device have an external SD card installed perhaps (or lack thereof)?

Also, do you need to share an image, or just a link?

**Comment by matuskopo** [comment]
Hi, thanks for fast response, finally I've found, I had no SD card plugged in. But then I have a question, if user has no SD card, this plugin will never work for him?

**Comment by EddyVerbruggen** [comment]
Hmm, I will do two things I guess (so reopening this issue):
- not create the dirs if no image is shared (I think it does at the moment).
- try to detect whether or not an SD card is installed.

**Comment by matuskopo** [comment]
I've already tested it, and if you send value 'null' instead of image URL, then everything works fine, even if you have no SD card installed. So there is problem only if you set any image URL. I had no time to test it with base64 image string.

**Comment by EddyVerbruggen** [comment]
That's good to hear, thanks for sharing your observations.

**Comment by abhishek97** [comment]
Any Update on this? , i is the same error with base64 image strtings  too.

**Comment by EddyVerbruggen** [comment]
Hey all, on Android when sharing an image (even if it's base64 encoded) the image needs to be written to a place where the receiving app can access it. That's an external directory which is on the user's SD card or on an external partition (Nexus devices have this).

I've updated the plugin to no longer crash in the (rare) case no external directory can be used, it then falls back to sharing other bits which may have been passed in (most relevant: a URL).

Available now in master, and in 4.3.20 when released.

I hope this makes sense,
Eddy
